### DISCUSSION TITLE
Should I run geosketch on samples separately?

### DISCUSSION THREAD

**Discussion by ayeTown** [brianhie/geosketch#14] [Q&A] [answered]
Hi @brianhie 
I have six samples from 3 WT mice and 3 KO mice. Should I run the algorithm on each sample separately to get a desired number of cells retained from each sample? Then combine the new samples together? Or should I run this on the full matrix (all mouse samples combined)? The reason I am asking is because I am interested in doing differential expression between WT vs. KO and I am worried running the algorithm on the full matrix may result in significant imbalance between the number of cells retained within each sample. Is there a way to run this to ensure there are n number of cells for each sample? Thanks for your time/help.
AT

**Comment by brianhie** [comment, marked as answer]
Hi @ayeTown,

My primary recommendation would be to do DE on the full data, without sketching. Many standard DE tests are efficient with the number of datapoints, so there shouldn't be any needed to perform sketching beforehand.

If you do need to downsample, I would uniformly downsample within each class, since one concern is that data-dependent downsampling will change the distribution of cells, leading to potentially misleading DE results. I think in the case of DE, you should probably just stick to more standard downsampling techniques.

**Comment by ayeTown** [comment]
Can you clarify what you mean by "within each class". Do you mean within each cell type? I was thinking of using geosketch for each sample ID (different biological replicate) separately so that the distribution of cell types would not change significantly within each sample.

**Reply by brianhie** [reply]
I meant uniform sampling the same number of cells within each biological replicate. The problem with using geosketch within each biological replicate is that it can change the distribution of cell types, whereas this is not a problem for uniform sampling.
